File: 20220629/sub.js
//배열 arr가 있다. 어떤 수 weight가 주어졌을 때 합쳐서 weight가 되는
//배열 내 두개의 인덱스 반환해라
//만약 합쳐서 weight가 되는 항목 두개가 존재하지 않을 경우 -1 return

const arr = [1, 2, 3, 4, 5];
const weight = 10;

for (let i = 0; i < arr.length; i++) {
  for (let k = i + 1; k < arr.length; k++) {
    if (arr[i] + arr[k] == weight) {
      console.log(i, k);
    } else {
      console.log(-1);
    }
  }
}

File: 20220727/App2/App/src/sub.java

class Sub {
    int[] subArr = new int[9];

    // {5,8,9,10} 5
    int[] divisible(int[] arr, int divisor) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] % divisor == 0) {
                subArr[i] = arr[i];
            }
        }
        return subArr;
    }
}

public class sub {
    public static void main(String[] args) {

        // 가운데 글자만 리턴하기
        // Ex) input Java = av
        // power = w

        int[] Num = new int[9];
        for (int i = 0; i < Num.length; i++) {
            Num[i] = (int) (Math.random() * 10);
        }

        Sub sub1 = new Sub();
        sub1.divisible(Num, 3);
        System.out.println(sub1.subArr);
    }

}
